exercises/all-your-base/test/test_all_your_base.c

```diff
@@ -0,0 +1,200 @@
+#include "vendor/unity.h"
+#include <stddef.h>
+#include <stdint.h>
+#include "../src/all_your_base.h"
+
+#include <stdio.h>
+
+void fill_array(int16_t data[], size_t data_length, int16_t array[],
+                size_t array_length)
+{
+   /* array expected to be longer than data */
+   for (size_t i = 1; i <= data_length; ++i)
+      array[array_length - i] = data[data_length - i];
+}
+
+void test_rebase(uint16_t input_base, int16_t input_digits[],
+                 size_t input_length, uint16_t output_base,
+                 int16_t expected_digits[], size_t expected_length)
+{
+   int16_t digits[DIGITS_ARRAY_SIZE] = { 0 };
+   int16_t ex_digits[DIGITS_ARRAY_SIZE] = { 0 };
+
+   fill_array(input_digits, input_length, digits, DIGITS_ARRAY_SIZE);
+   fill_array(expected_digits, expected_length, ex_digits, DIGITS_ARRAY_SIZE);
+
+   size_t actual_length = rebase(digits, input_base, output_base, input_length);
+
+   TEST_ASSERT_EQUAL_INT32(expected_length, actual_length);
+
+   if (expected_length > 0)
+      TEST_ASSERT_EQUAL_INT16_ARRAY(ex_digits, digits, expected_length);
+}
+
+void test_single_bit_to_decimal(void)
+{
+   int16_t input[] = { 1 };
```

Comment:
If you did:

```c
int16_t input[DIGITS_ARRAY_SIZE] = { 1 };
```

then it would automatically 0 fill the rest of the elements (making `fill_array` unnecessary).

Even better, this would allow you to get rid of the array length arguments to `test_rebase`. If we don't go this route, we should at least use `sizeof` and `sizeof(arr[0])` to calculate the number of elements instead of having them be hand-calculated.